src/main/java/co/cask/gcp/spanner/common/SpannerUtil.java

```diff
@@ -71,4 +71,83 @@ public static void validateSchema(Schema schema) {
       }
     }
   }
+
+  /**
+   * Converts schema to Spanner create statement
+   * @param tableName table name to be created
+   * @param schema schema of the table
+   * @return Create statement
+   */
+  public static String convertSchemaToCreateStatement(String tableName, Schema schema) {
+    StringBuilder createStmt = new StringBuilder();
+    createStmt.append("CREATE TABLE ").append(tableName).append(" (");
+
+    for (Schema.Field field : schema.getFields()) {
+      String name = field.getName();
+      Schema fieldSchema = field.getSchema();
+      fieldSchema = fieldSchema.isNullable() ? fieldSchema.getNonNullable() : fieldSchema;
+      Schema.LogicalType logicalType = fieldSchema.getLogicalType();
+      String spannerType;
+
+      if (logicalType != null) {
+        switch (logicalType) {
+          case DATE:
+            spannerType = "DATE";
+            break;
+          case TIMESTAMP_MILLIS:
+          case TIMESTAMP_MICROS:
+            spannerType = "TIMESTAMP";
+            break;
+          default:
+            // this should not happen
+            throw new IllegalStateException("Logical type" + logicalType + " is not supported.");
+        }
+        addColumn(createStmt, name, field.getSchema().isNullable(), spannerType);
+        continue;
+      }
+
+      Schema.Type type = fieldSchema.getType();
+      switch (type) {
+        case BOOLEAN:
+          spannerType = "BOOLEAN";
+          break;
+        case STRING:
+          spannerType = "STRING(MAX)";
+          break;
+        case LONG:
+          spannerType = "INT64";
+          break;
+        case DOUBLE:
+          spannerType = "FLOAT64";
+          break;
+        case BYTES:
+          spannerType = "BYTES(MAX)";
+          break;
+        default:
+          throw new IllegalStateException(type.name() + " : Type currently not supported.");
+      }
+      addColumn(createStmt, name, field.getSchema().isNullable(), spannerType);
+    }
+
+    // remove trailing ", "
+    createStmt.deleteCharAt(createStmt.length() - 1)
+      .deleteCharAt(createStmt.length() - 1)
+      .append(")");
+
+    // If table does not exist, by default first column becomes primary key
```

Comment:
The key should be specified by the user, rather than defaulted to the first field. Most of the time that won't be correct. 

I think we need to add a key property where the user can configure a list of fields for the key.